blockly/src/utils/BlocklyCommands.java

```diff
@@ -186,29 +188,35 @@ public static void pull() {
    * Shoots a fireball in direction the hero is facing.
    *
    * @param ac AmmunitionComponent of the hero, ammunition amount will be reduced by 1
+   * @param hero Entity to be used as hero for positioning
    */
-  private static void aimAndShoot(AmmunitionComponent ac) {
-    Entity hero = Game.hero().orElseThrow(MissingHeroException::new);
-    Direction viewDirection =
-        Direction.fromPositionCompDirection(EntityUtils.getViewDirection(hero));
-    Skill fireball =
-        new Skill(
-            new FireballSkill(
-                () ->
-                    hero.fetch(CollideComponent.class)
-                        .map(cc -> cc.center(hero))
-                        .map(p -> p.add(viewDirection.toPoint()))
-                        .orElseThrow(
-                            () -> MissingComponentException.build(hero, CollideComponent.class)),
-                FIREBALL_RANGE,
-                FIREBALL_SPEED,
-                FIREBALL_DMG),
-            1);
-    fireball.execute(hero);
+  private static void aimAndShoot(AmmunitionComponent ac, Entity hero) {
+    newFireballSkill(hero).execute(hero);
     ac.spendAmmo();
     Server.waitDelta();
   }
 
+  /**
+   * Create a new fireball for the given entity.
+   *
+   * @param hero Entity to be used as hero for positioning
+   * @return Nice new fireball, ready to be launched.
+   */
+  private static Skill newFireballSkill(Entity hero) {
```

Comment:
Frage mich, ob das nicht auch analog zu den Factory-Klassen irgendwo zentral gebaut werden kann?

Reply:
Ja so eine Skill- oder noch besser FireballFactory wäre schon praktisch, wir haben mal öfters das gleiche Pattern um nur einen Feuerball zu erstellen.

Reply:
@Flamtky War vorher nur ne Vermutung von mir - aber wenn das wirklich so ist, dann lass uns dafür doch eine Factory basteln? Machst Du ein Ticket auf, das kann ja eine der SHKs mal in Ruhe machen (oder später die drei Jungs in der Praxisphase)?

Reply:
Ich nehme an das ist #2006